src/components/colorscale/calc.js

```diff
@@ -74,32 +41,24 @@ module.exports = function calc(gd, trace, opts) {
         max += 0.5;
     }
 
-    doUpdate(minAttr, min);
-    doUpdate(maxAttr, max);
+    container['_' + minAttr] = container[minAttr] = min;
+    container['_' + maxAttr] = container[maxAttr] = max;
 
+    // TODO ?!?
     /*
      * If auto was explicitly false but min or max was missing,
      * we filled in the missing piece here but later the trace does
      * not look auto.
      * Otherwise make sure the trace still looks auto as far as later
      * changes are concerned.
      */
-    doUpdate(autoAttr, (auto !== false || (min === undefined && max === undefined)));
+    // doUpdate(autoAttr, (auto !== false || (min === undefined && max === undefined)));
```

Comment:
Could we remove this line?

Reply:
Good eye. Thanks!

Reply:
Fixed in https://github.com/plotly/plotly.js/pull/3341/commits/73c66a9a57b343a96ccc11bf3908d70fde5838e4